pkg/dependency/parser/java/pom/pom.go

```diff
@@ -303,7 +303,7 @@ func (d pomDependency) ToArtifact(opts analysisOptions) artifact {
 		Exclusions:   exclusions,
 		Locations:    locations,
 		Relationship: ftypes.RelationshipIndirect, // default
-		Test:         d.Scope == "test",
+		Test:         d.Scope == "test" || opts.testScope,
```

Comment:
I'm wondering if it works as expected. For example, if two direct dependencies have the same transitive dependency, and one scope is "test" and the other is not, I guess it would not be considered devDependency. How does Maven work?

Reply:
hm... Nice catch.

I [updated logic](https://github.com/aquasecurity/trivy/pull/7486/commits/51c9d3736e93c2c7416fd90d7897e7fdc1331913) and [added test for this case](https://github.com/aquasecurity/trivy/pull/7486/commits/6deda97b34a727c78c798d8162fcff000c1230f5) (comment before test shows how `mvn` works).
Can you take a look and check that this is what you meant?